core/sequencer/sequencer.go

```diff
@@ -72,24 +76,38 @@ func PeriodicallyRun(ctx context.Context, tickch <-chan time.Time, f func(ctx co
 	}
 }
 
+// TrackMasterships monitors resources for mastership.
+func (s *Sequencer) TrackMasterships(ctx context.Context) {
+	s.tracker.Run(ctx)
+}
+
 // RunBatchForAllDirectories scans the directories table for new directories and creates new receivers for
 // directories that the sequencer is not currently receiving for.
 func (s *Sequencer) RunBatchForAllDirectories(ctx context.Context) error {
 	directories, err := s.directories.List(ctx, false)
 	if err != nil {
 		return fmt.Errorf("admin.List(): %v", err)
 	}
-	// TODO(#565): Implement per-directory leader election here.
 	for _, d := range directories {
 		knownDirectories.Set(1, d.DirectoryID)
+		s.tracker.AddResource(d.DirectoryID)
+	}
```

Comment:
What happens to the deleted resources? Will `RunBatch` continuously return errors, resulting in skipping directory IDs that follow the deleted one in the loop below?

Reply:
Excellent catch.  I'll change that to a log and continue. 
I'm not sure what the best way to track deleted things is.